### PR TITLE
Implement and Test NFR #10866

### PR DESCRIPTION
**Pull request by lajosbencz**
NFR https://github.com/phalcon/cphalcon/issues/10866

### CODE REVIEW

**Comment by andresgutierrez** [comment]
This looks too arbitrary, not sure if everyone wants to ignore empty strings

File: phalcon/dispatcher.zep [hunk 1]
```diff
@@ -265,7 +265,11 @@ abstract class Dispatcher implements DispatcherInterface, InjectionAwareInterfac
 			this->{"_throwDispatchException"}("A dependency injection object is required to access the 'filter' service", self::EXCEPTION_NO_DI);
 		}
 		let filter = <FilterInterface> dependencyInjector->getShared("filter");
-		return filter->sanitize(paramValue, filters);
+		let paramValue = filter->sanitize(paramValue, filters);
+		if strlen(paramValue) < 1 && defaultValue !== null {
+			return defaultValue;
```
Comment (sergeyklay) [inline phalcon/dispatcher.zep]: Mb 
```zephir
return filter->sanitize(defaultValue, filters);
```
?

Comment (lajosbencz) [inline phalcon/dispatcher.zep]: Makes sense!

**Comment by lajosbencz** [comment]
Imho, I can't really think of a useful case when a 0 length dispatcher param is useful after sanitation. One will most likely handle it somehow, or just skip a prog branch. I think it's counter-intuitive if one still gets an empty string if a defaultValue is provided.
If only sanitation is the objective, defaultValue can be left empty.
I'm not keen on this implementation, because I don't need it in any project atm, it just makes sense this way :)

**Comment by lajosbencz** [comment]
On second thought, adding a ```hasParam``` method would also allow for more flexibility without any impairment

**Pull request closed without merging**